src/lib/containers/Facets.js

```diff
@@ -14,28 +22,85 @@ const DESELECT_ALL = "deselect all"
  */
 class CheckboxGroup extends React.Component {
 
+    constructor() {
+        super()
+        this.handleHover = this.handleHover.bind(this)
+        this.handleHoverExit = this.handleHoverExit.bind(this)
+    }
+
+    handleHover(event) {
+        if (event.target.tagName === "SPAN") {
+            event.target.className += " SRC-primary-background-color"
+        }
+    }
+    
+    handleHoverExit(event) {
+        if (event.target.tagName === "SPAN") {
+            let className = event.target.className
+            if (className.indexOf(" SRC-primary-background-color") !== -1 ) {
+                className = className.substring(0, className.indexOf(" SRC-primary-background-color"))
+                event.target.className = className
+            }
+        }
+    }
+
     render() {
         const {element} = this.props
         let children = []
-        let selectedFacets = this.props.selectedFacets
+
+        element.facetValues.sort((a,b) => {return b.count - a.count})
 
         element.facetValues.forEach(
             (facetValue, index) => {
                 let uniqueId = element.columnName + " " + facetValue.value + " " + facetValue.count
                 // caution when using uuId's to not cause extra re-renders from this always changing
-                let uuId = uuidv4()
+                let newR = this.props.RGB[0] * (1.3 - (1.0 / (index + 1)))
```

Comment:
how many facet values does it take for this algorithm to level out pure white (or black)?

Reply:
As of the moment there is always going to be an almost pure black color chosen (index === 0 will set each RGB components to near zero which is why I used 1.3 to offset it from being pure black). Pure white isn't achievable with this algorithm, as index gets really large the original color will be chosen (it wont the the exact original color b/c of the 1.3 but close..). So I guess the color spectrum here is [black, original color chosen * 1.3]

So to answer the question:  pure black- 1 facet value, pure white- N/A

 I saw other ideas online that I can try for this too.